### PR TITLE
Do not return value from "cleanup" functions

### PR DESCRIPTION
**Pull request by jugglinmike**
Today, the return value of functions provided to the global
`add_cleanup` function has no effect on the behavior of the test runner.
An upcoming feature addition to `testharness.js` will cause the return
value to influence test results [1].

Despite this, some existing tests have already been authored to return a
value. Introducing the new test harness functionality may trigger
unexpected results in those contexts. To avoid this, refactor existing
tests to avoid returning a value.

[1] https://github.com/w3c/web-platform-tests/issues/6075

---

The Service Worker test I've modified returns a "thenable" which is actually
amenable to the proposed `testharness.js` extension. We could leave it as-is,
and when/if an implementation for gh-6075 lands, that test would become less
fragile automatically. Still, it seems safer to introduce the new functionality
with the confidence that it doesn't trigger *any* implicit opt-in, and then
incrementally begin to use it with intentional follow-up commits. (And for the
Service Worker test suite, this will entail updating many files besides the one
modified here.)

For those curious: I located these tests programatically by [modifying the test
runner](https://github.com/w3c/web-platform-tests/compare/w3c:de65155...bocoup:8fe4fe1) and using that to run all tests that reference `add_cleanup`:

    grep 'add_cleanup' . -rl --include '*.js' --include '*.html' | xargs ./wpt run chrome

And

    grep 'add_cleanup' . -rl --include '*.js' --include '*.html' | xargs ./wpt run firefox

Both browsers identified the same usages.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21997)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/318836119)

File: service-workers/service-worker/postmessage.https.html [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
     return service_worker_unregister_and_register(t, script, scope)
       .then(r => {
-          t.add_cleanup(() => r.unregister());
+          t.add_cleanup(() => { r.unregister(); });
```
Comment (foolip) [inline service-workers/service-worker/postmessage.https.html]: Can you add a TODO here to revert it back when promise cleanup has landed?

**Review by foolip — APPROVED** [review]
LGTM % tracking the remaining work.

File: service-workers/service-worker/postmessage.https.html (hunk 1 above)
Comment (jugglinmike) [inline service-workers/service-worker/postmessage.https.html]: Sure thing

**Review by foolip — APPROVED** [review]

**Pull request merged**